main.py

```diff
@@ -138,6 +138,10 @@ def log_startup_message(port_num, is_https_enabled):
                 SSL_CERT = sys.argv[i+1]
             elif (arg == "--ssl-key" and (i+1) < (len(sys.argv))):
                 SSL_KEY = sys.argv[i+1]
+            elif (arg == "--port" and (i+1) < (len(sys.argv))):
+                port = int(sys.argv[i+1])
```

Comment:
Please handle converting string to int and avoid runtime issues. You can catch it gracefully, log, and exit.